compiler/shared_ast/typing.ml

```diff
@@ -282,10 +309,18 @@ let polymorphic_op_type (op : Operator.polymorphic A.operator Mark.pos) :
     | Log (PosRecordIfTrueBool, _) -> [bt] @-> bt
     | Log _ -> [any] @-> any
     | Length -> [array any] @-> it
-    | HandleDefault -> [array ([ut] @-> any); [ut] @-> bt; [ut] @-> any] @-> any
+    | HandleDefault ->
+      [array ([ut] @-> default any); [ut] @-> bt; [ut] @-> any] @-> default any
     | HandleDefaultOpt ->
-      [array (option any); [ut] @-> option bt; [ut] @-> option any]
-      @-> option any
+      (* FIXME: when translating with avoid_exceptions, defaults morally become
+         options everywhere. However, this is not reflected in the environment
+         at the moment, so this operator may have to mix-and-match options and
+         default terms ; since the typing of default is expected to guide a
+         simplification of the avoid-exceptions backend anyway, this is left
+         untyped at the moment *)
+      any
```

Comment:
Wait if there's this mix and match at the moment, how are you able to type output OCaml programs compiled without exceptions?

Reply:
the mix-and-match only concerns the inputs of the operator that are either of type option, or default. But @adelaett will look into it, it should definitely not stay like that